Versioner/PrinceOfVersions.swift

```diff
@@ -103,5 +119,69 @@ public struct PrinceOfVersions {
             }
         })
     }
+}
 
+extension PrinceOfVersions: URLSessionDelegate {
+    
+    public func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+        
+        guard let trust = challenge.protectionSpace.serverTrust, SecTrustGetCertificateCount(trust) > 0 else {
+            completionHandler(.cancelAuthenticationChallenge, nil)
+            return
+        }
+        
+        guard _pinnedCertificateURL != nil else {
+            completionHandler(.performDefaultHandling, nil)
+            return
+        }
+        
+        if let serverCertificate = SecTrustGetCertificateAtIndex(trust, 0), let serverCertificateKey = publicKey(for: serverCertificate) {
+            
+            let hasKey = pinnedKeys().contains(where: { (key) -> Bool in
+                if (serverCertificateKey as AnyObject).isEqual(key) {
+                    return true
+                } else {
+                    return false
+                }
+            })
+            
+            if hasKey {
+                completionHandler(.useCredential, URLCredential(trust: trust))
+                return
+            }
+        }
+        
+        completionHandler(.cancelAuthenticationChallenge, nil)
+    }
+    
+    fileprivate func pinnedKeys() -> [SecKey] {
```

Comment:
Private methods should have underscore prefix.

Reply:
Fixed.